src/config/src/utils/flatten.rs

```diff
@@ -139,17 +139,37 @@ pub fn format_key(key: &mut String) {
     if check_key(key) {
         return;
     }
-    let mut key_chars = key.chars().collect::<Vec<_>>();
-    for c in key_chars.iter_mut() {
-        if c.is_lowercase() || c.is_numeric() {
-            continue;
-        } else if c.is_uppercase() {
-            *c = c.to_lowercase().next().unwrap();
+
+    let bytes = unsafe { key.as_bytes_mut() };
+    let mut i = 0;
+
+    while i < bytes.len() {
+        if bytes[i].is_ascii() {
+            if bytes[i].is_ascii_uppercase() {
+                bytes[i] = bytes[i].to_ascii_lowercase();
+            } else if !bytes[i].is_ascii_lowercase()
+                && !bytes[i].is_ascii_digit()
+                && bytes[i] != b'_'
+            {
+                bytes[i] = b'_';
+            }
         } else {
-            *c = '_';
+            *key = key
+                .chars()
+                .map(|c| {
+                    if c.is_lowercase() || c.is_numeric() {
+                        c
+                    } else if c.is_uppercase() {
+                        c.to_lowercase().next().unwrap()
+                    } else {
+                        '_'
+                    }
+                })
+                .collect();
```

Comment:
This PR looks great, only this place if it is not ascii, will rebuild the `key` multiple times. and if in this section we replaced one char, i am not sure if it still work for next ascii.

Reply:
np, it will return once we found none-asscii char.